cs2/src/state/player.rs

```diff
@@ -116,6 +120,30 @@ impl State for StatePawnInfo {
 
         let player_flashtime = player_pawn.m_flFlashBangTime()?;
 
+        let player_has_bomb = entities

```

Comment:
Iterating over all entities every time you generate the player pawn info seems rather costly. I would recommend factoring  the "BombPlayerInfo" into a state which can be cached and only needs to be resolved once a render cycle.